api/src/main/java/com/epam/pipeline/manager/git/GitManager.java

```diff
@@ -879,6 +898,121 @@ public GitProject copyRepository(final String projectName, final String newProje
         return forkedProject;
     }
 
+    public GitReaderEntryListing<GitRepositoryEntry> lsTreeRepositoryContent(final Long id, final String version,
+                                                                             final String path, final Long page,
+                                                                             final Integer pageSize) {
+        try {
```

Comment:
I think we can introduce some common method to call GitReaderClient methods. e.g. create a parametrized method:
```
private T <T> callGitReaderApi(final Long id, 
    final String version, 
    Function<GitReaderClient, T> method) {
    try {
           //load pipeline and validate revision
          return method.apply(getGitReaderHostPreference());
    } catch {
          ....
    }
}
```
and call it:
```
    lsTreeRepositoryContent(final Long id, final String version,
                            final String path, final Long page,
                            final Integer pageSize) {
         return callGitReaderApi(id,  version, (reader)  -> reader. getRepositoryTree(path, version, ...));
    }
```